encoder/vaapiencoder_h264.cpp

```diff
@@ -1314,6 +1511,214 @@ bool VaapiEncoderH264::fillReferenceList(VAEncSliceParameterBufferH264* slice) c
     return true;
 }
 
+bool VaapiEncoderH264::addPackedPrefixNalUnit(const PicturePtr& picture) const
+{
+    bool ret = true;
+    BitWriter bs;
+    bs.writeBits(H264_NAL_START_CODE, 32);
+    bit_writer_write_nal_header(&bs, picture->m_isReference
+                                         ? VAAPI_ENCODER_H264_NAL_REF_IDC_LOW
+                                         : VAAPI_ENCODER_H264_NAL_REF_IDC_NONE,
+                                VAAPI_ENCODER_H264_NAL_PREFIX);
+
+    bs.writeBits(1, 1); /* svc_extension_flag */
+    bs.writeBits(picture->isIdr(), 1);
+    bs.writeBits(picture->m_priorityId, 6);
+    bs.writeBits(1, 1); /* no_inter_layer_pred_flag */
+    bs.writeBits(0, 3); /* dependency_id */
+    bs.writeBits(0, 4); /* quality_id */
+    bs.writeBits(picture->m_temporalId, 3);
+    bs.writeBits(0, 1); /* use_ref_base_pic_flag */
+    bs.writeBits(1, 1); /* discardable_flag */
+    bs.writeBits(1, 1); /* output_flag */
+    bs.writeBits(3, 2); /* reserved_three_2bits */
+
+    if (picture->m_isReference) {
+        bs.writeBits(0, 1); /* store_ref_base_pic_flag */
+        bs.writeBits(0, 1); /* additional_prefix_nal_unit_extension_flag*/
+    }
+    /* no more rbsp data */
+
+    bit_writer_write_trailing_bits(&bs);
+
+    uint8_t* codedData = bs.getBitWriterData();
+    ASSERT(codedData);
+
+    if (!picture->addPackedHeader(VAEncPackedHeaderRawData, codedData,
+                                  bs.getCodedBitsCount())) {
+        ret = false;
+    }
+
+    return ret;
+}
+
+bool VaapiEncoderH264::addPackedSliceHeader(
+    const PicturePtr& picture,
+    const VAEncSliceParameterBufferH264* const sliceParam) const
```

Comment:
it is not to let encoder driver know something. We write it to bitstream and let decoder know to how to reordered.